plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -0,0 +1,197 @@
+            builder.put(column, parsedValue);
+        }
+        Map<ColumnHandle, NullableValue> values = builder.buildOrThrow();
+        return new HivePartition(tableName, partitionName, values);
+    }
+
+    public static boolean partitionMatches(List<HiveColumnHandle> partitionColumns, TupleDomain<HiveColumnHandle> constraintSummary, HivePartition partition)
+    {
+        if (constraintSummary.isNone()) {
+            return false;
+        }
+        Map<HiveColumnHandle, Domain> domains = constraintSummary.getDomains().orElseGet(ImmutableMap::of);
+        for (HiveColumnHandle column : partitionColumns) {
+            NullableValue value = partition.getKeys().get(column);
+            Domain allowedDomain = domains.get(column);
+            if (allowedDomain != null && !allowedDomain.includesNullableValue(value.getValue())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static List<HivePartitionKey> buildPartitionKeys(List<Column> keys, List<String> values)
+    {
+        checkCondition(keys.size() == values.size(), HIVE_INVALID_METADATA,
+                "Expected %s partition key values, but got %s. Keys: %s, Values: %s.",
+                keys.size(), values.size(), keys, values);
+        ImmutableList.Builder<HivePartitionKey> partitionKeys = ImmutableList.builder();
+        for (int i = 0; i < keys.size(); i++) {
+            String name = keys.get(i).getName();
+            String value = values.get(i);
+            partitionKeys.add(new HivePartitionKey(name, value));
+        }
+        return partitionKeys.build();
+    }
+
+    public static PartitionValueExtractor inferPartitionValueExtractor(
+            String relativePartitionPath,
+            List<String> expectedPartitionValues)
+            throws HoodieIOException
+    {
+        for (PartitionValueExtractor partitionValueExtractor : PARTITION_VALUE_EXTRACTORS) {
+            try {
+                List<String> extractedPartitionValues =
+                        partitionValueExtractor.extractPartitionValuesInPath(relativePartitionPath);
+                if (extractedPartitionValues.equals(expectedPartitionValues)) {
+                    log.debug("Inferred %s to be the partition value extractor",
+                            partitionValueExtractor.getClass().getName());
+                    return partitionValueExtractor;
+                }
+                else {
+                    log.debug("Cannot use partition value extractor %s due to value mismatch (expected: %s, actual: %s), trying the next option ...",
+                            partitionValueExtractor.getClass().getName(),
+                            expectedPartitionValues,
+                            extractedPartitionValues);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                log.debug("Cannot use partition value extractor %s, trying the next option ...",
```

Comment:
Is swallowing the exception here legitimate?

Reply:
Yes, this is Hudi-specific. Hudi allows custom partition value extraction logic. It's ok to swallow the exception here.